### DISCUSSION TITLE
create dependencies among argo CronWorkflow

### DISCUSSION THREAD

**Discussion by baljijit** [argoproj/argo-workflows#12241] [Q&A] [answered]
Hi,
is there a way to create dependencies among two or more argo workflows? In my case, i have two CronWorkflow and i would like to start the second workflow if the first finish well. 
Obviously, there is a temporal dependency created implicitly by schedule time like first has to start at 01:00 and the second at 02:00. What i would like to do is the second one, just, if first finish without errors.
Thanks in advance!

**Comment by 0xdarkman** [comment]
You can create cronworkflows in airlfow which are DAGs running on schedule and build dependency between them with marker/sensor.
You keep DAGs independent from each other while still embed some dependency on e.g. upstream DAG, upstream DAG task state. 
I would like to see similar solution in Argo Workflows which would enable to build Cronworkflow to Cronworkflow dependency without building dag of dags.

**Comment by 0xdarkman** [comment]
My work around for argo workflows is:

each Cronworkflow writes with appender task (last task) to delta table the status of the executon when completes e.g. name, status, metadata (timestamp and some other data).

each Cronworkflows runs scheduler as first task which reads table and checks status of relevant upstreams Cronworkflows execution completion state and timestamps. It checks with retry and backoff streategy in order to encounter for longer durations of upstream executions than usually. 

this mimics for me airflow marker/sensor approach when I use this table as the workaround.

**Reply by agilgur5** [reply]
I think this is the correct approach, as it is independent of your tooling and has knowledge of your tables and upstreams, making it more resilient and accurate.

**Comment by agilgur5** [comment, marked as answer]
To answer the question directly, you can use a [`resource` template](https://argo-workflows.readthedocs.io/en/latest/walk-through/kubernetes-resources/) to poll the previously created Workflow for its `status`. Could also use a `script` template to directly run a `kubectl get`.

But a more resilient approach is written [above](https://github.com/argoproj/argo-workflows/discussions/12241#discussioncomment-9223203).
